### PR TITLE
Add meta registration for bucketize

### PR DESCRIPTION
**Pull request by suo**

### CODE REVIEW

**Review by zhxchen17 — APPROVED** [review]

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -2575,6 +2575,11 @@ def mkldnn_rnn_layer_backward(
     return diff_x, diff_w1, diff_w2, diff_b, diff_b, diff_hx, diff_cx
 
 
+@register_meta(aten.bucketize)
+def meta_bucketize(a, boundaries, *, out_int32=False, right=False):
+    return a
```
Comment (SherlockNoMad) [inline torch/_meta_registrations.py:2580]: `return a` if only the aten op is an inplace op.
Otherwise, you need to `return torch.empty_like(self)`

Comment (SherlockNoMad) [reply torch/_meta_registrations.py:2580]: as meta function need to correctly propagate "aliasing relationship"

File: torch/_meta_registrations.py [hunk 2]
```diff
@@ -2575,6 +2575,11 @@ def mkldnn_rnn_layer_backward(
     return diff_x, diff_w1, diff_w2, diff_b, diff_b, diff_hx, diff_cx
 
 
+@register_meta(aten.bucketize)
```
Comment (SherlockNoMad) [inline torch/_meta_registrations.py:2578]: This impl only works for aten.bucketize.Tensor, and won't work for aten.bucketize.Scalar.

as torch.empty_like(scalar) doesn't work.... 

You might want to update it to `@register_meta(aten.bucketize.Tensor)`

File: torch/_meta_registrations.py [hunk 3]
```diff
@@ -2575,6 +2575,11 @@ def mkldnn_rnn_layer_backward(
...
+def meta_bucketize(a, boundaries, *, out_int32=False, right=False):
```
Comment (SherlockNoMad) [inline torch/_meta_registrations.py:2579]: also need to change dtype according to out_int32

File: torch/_meta_registrations.py [hunk 4]
```diff
@@ -2575,6 +2575,11 @@ def mkldnn_rnn_layer_backward(
...
+    return torch.empty_like(a)
```
Comment (SherlockNoMad) [inline torch/_meta_registrations.py:2580]: return dtype can only be int32 or int64.

So it's more like `torch.empty_like(a, dtype=torch.int32 if out_int32 else torch.int64`

**Comment by Chillee** [comment]
Shouldn't you change test_meta?

**Comment by suo** [comment]
I dunno, I guess I should lol

**Pull request closed without merging**